packages/@expo/metro-config/src/ExpoMetroConfig.ts

```diff
@@ -283,6 +331,21 @@ export function getDefaultConfig(
           require.resolve(path.join(reactNativePath, 'Libraries/Core/InitializeCore')),
         ];
 
+        // Add the initialize core for OOT platforms
+        if (customPlatforms) {
+          for (const platformName in customPlatforms) {
+            const initializeFile = `${customPlatforms[platformName]}/Libraries/Core/InitializeCore`;
```

Comment:
`getModulesRunBeforeMainModule` is a bad hack, we should just add these files somewhere in the import chain.